### PR TITLE
"Modernise" the build process and fix a few related bugs

### LINKED ISSUE #434: python setup.py bdist_wheel raises `TypeError: expected string or bytes-like object`
I'm running on python3.8.10 and other softwares are latest version
```
/usr/lib/python3/dist-packages/setuptools/dist.py:481: UserWarning: The version specified (<src.drozer.meta.Version object at 0x7f37bbd37fa0>) is an invalid version, this may not work as expected with newer versions of setuptools, pip, and PyPI. Please see PEP 440 for more details.
  warnings.warn(
running bdist_wheel
Traceback (most recent call last):
  File "setup.py", line 125, in <module>
    setuptools.setup(
  File "/usr/lib/python3/dist-packages/setuptools/__init__.py", line 144, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.8/distutils/dist.py", line 984, in run_command
    cmd_obj.ensure_finalized()
  File "/usr/lib/python3.8/distutils/cmd.py", line 107, in ensure_finalized
    self.finalize_options()
  File "/usr/lib/python3/dist-packages/wheel/bdist_wheel.py", line 129, in finalize_options
    self.data_dir = self.wheel_dist_name + '.data'
  File "/usr/lib/python3/dist-packages/wheel/bdist_wheel.py", line 164, in wheel_dist_name
    safer_version(self.distribution.get_version()))
  File "/usr/lib/python3/dist-packages/wheel/bdist_wheel.py", line 43, in safer_version
    return safe_version(version).replace('-', '_')
  File "/usr/lib/python3/dist-packages/pkg_resources/__init__.py", line 1332, in safe_version
    return str(packaging.version.Version(version))
  File "/usr/lib/python3/dist-packages/pkg_resources/_vendor/packaging/version.py", line 200, in __init__
    match = self._regex.search(version)
TypeError: expected string or bytes-like object
```

### LINKED ISSUE #439: Review the old build process
Our current build process is a mess. It doesn't correctly declare executable scripts (especially impactful on Windows, since we basically need a `.bat` file to run Python lmao), it relies on running `setup.py` as a script ([apparently a no-no](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/)), it introduces [silly bugs](https://github.com/WithSecureLabs/drozer/issues/434), and makes working with IDEs clunky.

Probably needs a rewrite as a TOML config file for setuptools. Leaving this one to Alex because they know what they're doing

### PR DESCRIPTION
**Pull request by nyaalex**
In short the `setup.py` file has been replaced with a `pyproject.toml`. On top of this the overall process has been modified to be more in line with python packaging best practices, and to improve cross-compatibility.

### CODE REVIEW

**Comment by cyberMilosz** [comment]
Don't merge just yet - a couple more tweaks to `README.md` incoming

**Comment by cyberMilosz** [comment]
Should be good to go
Tested on Windows (woooo we support Windows now), a couple of Linux builds an macOS

**Review by cyberMilosz — APPROVED** [review]

**Review by Yogehi — APPROVED** [review]
It's 2am where I'm at and I'm about to leave for Japan.

After 10 seconds of skimming, approved lol

**Pull request merged**
